### PR TITLE
Create ideas bank page

### LINKED ISSUE #537: Funcionalidades Banco de Idéias
## Feature Description

O banco de ideias é uma funcionalidade disponível para todos os usuários registrarem suas demandas de pesquisas e tecnologias. Vai ser a oportunidade para aqueles que não encontraram a tecnologia que buscam e sugerir que os pesquisadores desenvolvam de acordo com a demanda da sociedade.

O banco de ideias será composto por:

**1 - Lista corrida de oportunidades cadastradas pelos usuários**
- Uma pequena descrição abaixo do título da página:
_Não encontrou o que desejava?
Aqui você pode sugerir novas ideias de desenvolvimento de pesquisa e tecnologia de acordo com sua necessidade e para resolver problemas comuns do nosso povo. Veja abaixo as ideias já sugeridas pelos nossos usuários e registre a sua também._

- Lista simples composta pelo seguinte conteúdo
  - Título da ideia (texto)
  - Detalhe da ideia (textarea)
  - Palavras-chave
  - Usuário que sugeriu a ideia
  - Data do registro
- Incluir opções para filtrar a lista por texto do título e descrição e por palavra-chave
- Essa lista será acessar por meio do menu principal da Plataforma (estamos definindo esse menu ainda)

**2 - Cadastrar uma ideia**
- Somente usuários cadastrados podem registrar uma ideia
- Quando clicar em REGISTRAR UMA IDEIA aparecer uma tela com os seguintes campos:
- TÍTULO DA IDEIA
- DESCRIÇÃO DA IDEIA (textarea): 
- PALAVRAS-CHAVE (o mesmo do cadastro de tecnologias)
- botão ENVIAR IDEIA
 
**3 - Lista de Ideias para os usuários pesquisadores**
- Na área do pesquisador incluir um item no menu chamado BANCO DE IDEIAS
- Apresentar os seguintes campos na tabela:
  - Título
  - Usuário
  - Data
  - Palavras-chave
- Incluir opções para filtrar a lista por texto do título e descrição e por palavra-chave
- Incluir um ícone de detalhes que apresenta um modal com a descrição da ideia

---------------

## Acceptance criteria

* 
- Somente usuários logados e cadastrados poderão registrar novas ideias.

## Implementation Brief

*

### PR DESCRIPTION
**Pull request by calaca**
## Summary

Resolves #537 

## Relevant technical choices

- Environment variables were updated
- Algolia indexes for specific environments now end with said environment name (e.g. `searchable__technology__production`)

## QA Steps

1. Update your `.env` (see `.env.example`)
2. At home page, click `recursos > banco de ideias`
3. The page is responsive
4. It's possible to add a new idea
5. All ideas are listed and have instant search by Algolia

## Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code passes the linting.
- [x] My code has proper inline documentation.
- [x] I have manually tested this PR.

### CODE REVIEW

File: packages/web/utils/helper.js [hunk 1]
```diff
@@ -327,3 +327,26 @@ export const stringToLocaleDate = (date, options = {}) =>
  */
 export const stringToLocaleTime = (date, options = {}) =>
 	new Date(date).toLocaleTimeString('pt-br', { ...options });
+
+/**
+ * Handle phone number mask, necessary to alternate between masks when phone has 9 digits
```
Comment (luizeboli) [inline packages/web/utils/helper.js:332]: Is this the true description? I'm confused due it's the same of `beforeMaskedValueChange` but seems to have another purpose 😅

**Review by luizeboli — CHANGES_REQUESTED** [review]
Wow, awesome job @calaca 🎉 

I did some comments in order to improve usability.

1. What about adding a little bit more margin between these fields messages? So users better discern it.
image

2. After submitting a new idea only `Titúlo da ideia` input is cleared
image

3. I've run `npm run algolia:index` command as I'm in a new environment, but when sorting by update or release date this error occurs:
image

Looking at Algolia dashboard these indexes really doesn't exists. Maybe you forgot to add?

**Comment by calaca** [comment]
> Wow, awesome job @calaca 
> 
> I did some comments in order to improve usability.
> 
> 1. What about adding a little bit more margin between these fields messages? So users better discern it.
>    image
> 2. After submitting a new idea only `Titúlo da ideia` input is cleared
>    image
> 3. I've run `npm run algolia:index` command as I'm in a new environment, but when sorting by update or release date this error occurs:
>    image
> 
> Looking at Algolia dashboard these indexes really doesn't exists. Maybe you forgot to add?

Thanks for the review @luizeboli !

1. I've added that same input wrapper that exists around the title input to the other ones
2. I've managed to clear all other fields using `setValue` but it seems like there's a bug with `TextField` that applies the clean value then re-applies the old text. Maybe it should do into a separate issue as it happens at the separate contact page as well and every other place.
3. This error happens because these indexes don't exist in Algolia yet. Mateus said he'd be working on this as soon as he gets some time. But the front part of it is ready, we just need that data indexed by the API.

**Comment by luizeboli** [comment]
@calaca theres a `reset` method exposed by `useForm`, but to make it work properly you must pass `defaultValues` to the form. So you can call `form.reset`, but, we're using the `Form` component. I couldn't expose this reset method using `useImperativeHandle`, because it's mandatory to pass a ref, doing things this way would break other parts.

**Review by luizeboli — DISMISSED** [review]

File: packages/web/components/LandingPage/About/index.js [hunk 2]
```diff
@@ -1,22 +1,14 @@
...
-import {
-	Wrapper,
-	Container,
-	TextContainer,
-	Title,
-	TextContent,
-	TopImage,
-	LogosContainer,
-} from './styles';
+import * as S from './styles';
```
Comment (lcnogueira) [inline packages/web/components/LandingPage/About/index.js:3]: I prefer that way too nowadays 👌

**Review by lcnogueira — APPROVED** [review]

**Review by luizeboli — APPROVED** [review]

**Pull request merged**